protect_archiver/cli/download.py

```diff
@@ -1,111 +1,152 @@
+    show_default=True,
+    help="Skip downloading files which already exist on disk"
 )
 @click.option(
     "--touch-files",
-    default=False,
     is_flag=True,
+    default=False,
+    show_default=True,
     help=(
-        "Create local file without content for current download (Default: False) - "
-        "useful in combination with '--skip-existing-files' to skip problematic segments"
-    ),
+            "Create local file without content for current download - "
+            "useful in combination with '--skip-existing-files' to skip problematic segments"
+    )
 )
 @click.option(
     "--use-subfolders/--no-use-subfolders",
     default=True,
-    help="Save footage to folder structure with format 'YYYY/MM/DD/camera_name/' (Default: False)",
+    show_default=True,
+    help="Save footage to folder structure with format 'YYYY/MM/DD/camera_name/'"
 )
 @click.option(
     "--download-request-timeout",
     "download_timeout",
     default=60.0,
-    help="Time to wait before aborting download request, in seconds (Default: 60)",
+    show_default=True,
+    help="Time to wait before aborting download request, in seconds"
 )
 @click.option(
     "--start",
+    type=click.DateTime(formats=['%Y-%m-%d', '%Y-%m-%dT%H:%M:%S', '%Y-%m-%d %H:%M:%S', '%Y-%m-%d %H:%M:%S%z']),
     required=False,
-    help='Start time in dateutil.parser compatible format, for example "YYYY-MM-DD HH:MM:SS+0000"',
+    help=(
+            "Download range start time. "
+            # TODO(danielfernau): uncomment the next line as soon as the feature is implemented
+            # "If omitted, the time of the first available recording for each camera will be used."
+    ),
 )
 @click.option(
     "--end",
-    type=str,
+    type=click.DateTime(formats=['%Y-%m-%d', '%Y-%m-%dT%H:%M:%S', '%Y-%m-%d %H:%M:%S', '%Y-%m-%d %H:%M:%S%z']),
     required=False,
-    help='End time in dateutil.parser compatible format, for example "YYYY-MM-DD HH:MM:SS+0000"',
+    help=(
+            "Download range end time. "
+            # TODO(danielfernau): uncomment the next line as soon as the feature is implemented
+            # "If omitted, the time of the last available recording for each camera will be used."
+    ),
 )
 @click.option(
     "--snapshot",
     "create_snapshot",
-    default=False,
     is_flag=True,
-    help="Capture and download a snapshot from the specified camera(s)",
+    default=False,
+    show_default=True,
+    help=(
+            "Capture and download a snapshot from the specified camera(s). "
+            "This flag cannot be used in combination with the normal video download mode."
+    )
 )
 def download(
-    dest,
-    address,
-    port,
-    username,
-    password,
-    verify_ssl,
-    cameras,
-    download_wait,
-    download_timeout,
-    use_subfolders,
-    touch_files,
-    skip_existing_files,
-    ignore_failed_downloads,
-    start,
-    end,
-    create_snapshot,
+        dest,
```

Comment:
indentation here probably came from black fwiw

`black .` will run it on the codebase - we could also setup pre-commit if wanted to try to consistently do it

Reply:
Using `black` now, too. Thanks for the hint. Format should be as expected now 😄